File: naTanjir/naTanjir.Services/Database/StavkeNarudzbe.cs
using System;
using System.Collections.Generic;

namespace naTanjir.Services.Database;

public partial class StavkeNarudzbe:ISoftDelete
{
    public int StavkeNarudzbeId { get; set; }

    public int Kolicina { get; set; }

    public decimal Cijena { get; set; }

    public bool IsDeleted { get; set; }

    public DateTime? VrijemeBrisanja { get; set; }

    public int? NarudzbaId { get; set; }

    public int ProizvodId { get; set; }

    public int RestoranId { get; set; }

    public virtual Narudzba? Narudzba { get; set; }

    public virtual Proizvod Proizvod { get; set; } = null!;

    public virtual Restoran Restoran { get; set; } = null!;
}

File: naTanjir/UI/natanjir_mobile/lib/models/stavke_narudzbe.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:natanjir_mobile/models/korisnici.dart';
import 'package:natanjir_mobile/models/narudzba.dart';
import 'package:natanjir_mobile/models/proizvod.dart';
import 'package:natanjir_mobile/models/restoran.dart';

part 'stavke_narudzbe.g.dart';

@JsonSerializable()
class StavkeNarudzbe {
  int? stavkeNarudzbeId;
  int? kolicina;
  double? cijena;
  int? narudzbaId;
  int? proizvodId;
  int? korisnikId;
  int? restoranId;
  Proizvod? proizvod;
  Restoran? restoran;
  StavkeNarudzbe();

  factory StavkeNarudzbe.fromJson(Map<String, dynamic> json) =>
      _$StavkeNarudzbeFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$StavkeNarudzbeToJson(this);
}
